File: problems/Meeting Rooms II/minMeetingRooms.cs
public class Solution {
    public int MinMeetingRooms(int[][] intervals) {
        var n = intervals.Length;
        var starts = new int[n];
        var ends = new int[n];
        var sIdx = 0;
        var eIdx = 0;
        var currRoomsCount = 0;
        var roomsCount = 0;

        for (var i = 0; n > i; ++i) {
            starts[i] = intervals[i][0];
            ends[i] = intervals[i][1];
        }

        Array.Sort(starts);
        Array.Sort(ends);

        while (n > sIdx) {
            if (starts[sIdx] < ends[eIdx]) {
                roomsCount = Math.Max(roomsCount, ++currRoomsCount);
                ++sIdx;
            } else {
                --currRoomsCount;
                ++eIdx;
            }
        }

        return roomsCount;
    }
}

File: problems/Meeting Rooms II/minMeetingRooms.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        int result = 0;
        int n = intervals.length;
        int[] ends = new int[n];
        int[] starts = new int[n];
        int eIdx = 0;
        int sIdx = 0;
        int rooms = 0;

        for (int idx = 0; n > idx; ++idx) {
            ends[idx] = intervals[idx][1];
            starts[idx] = intervals[idx][0];
        }

        Arrays.sort(ends);
        Arrays.sort(starts);

        while (n > sIdx) {
            if (starts[sIdx] < ends[eIdx]) {
                result = Math.max(result, ++rooms);
                ++sIdx;
            } else {
                --rooms;
                ++eIdx;
            }
        }

        return result;
    }
}
